cli/runner/orchestrator.go

```diff
@@ -128,11 +127,17 @@ func (o orchestrator) Run(ctx context.Context, r Runner, opts RunOptions, output
 		zap.Strings("requiredGates", opts.RequiredGates),
 	)
 
-	varsID, err := o.resolveVarsID(ctx, opts.VarsID)
-	if err != nil {
-		return ExitCodeGeneralError, fmt.Errorf("cannot resolve variable set id: %w", err)
+	var err error
+
+	varsID := ""
+
+	if opts.VarsID != "" {
+		varsID, err = o.resolveVarsID(ctx, opts.VarsID)
+		if err != nil {
+			return ExitCodeGeneralError, fmt.Errorf("cannot resolve variable set id: %w", err)
+		}
+		o.logger.Debug("env resolved", zap.String("ID", varsID))
```

Comment:
Small check to avoid an unnecessary API call to the VariableSet API when the user passes no `VarID`.

Reply:
I think this would be easier to follow if the condition is on the `resolveVarsID` method

Reply:
Makes sense! I'll move this validation to`resolveVarsID `